samwise.js

```diff
@@ -1,6 +1,7 @@
 /**
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.